sources/api/migration/migrator/src/main.rs

```diff
@@ -309,29 +308,36 @@ where
             output.status.success(),
             error::MigrationFailureSnafu { output }
         );
+
+        // If an intermediate datastore exists from a previous loop, delete it.
+        delete_intermediate_datastore(&intermediate_datastore);
+        // Remember the location of the target_datastore to delete it in the next loop iteration
+        // (i.e if it was an intermediate).
+        intermediate_datastore = Some(target_datastore.clone());
```

Comment:
Won't this flow leave around the last migration's datastore? Or is that the desired behavior? From how I read this code, all migrations are executed, deleting the previous datastore as it goes, but on the last migration it deletes the previous datastore and stores the current target one. Then it exits the loop.

What am I missing here?